Use JS loader and export contents.

Zosoled opened the pull request:
Line breaks should be retained when either (a) not minifying, or (b) preserving legal comments. This switches the esbuild loader to `"js"` and creates the returned contents as a template literal instead of string. It also updates tests to conform to the template literal implementation.

vanruesc committed:
Update ci.yml

vanruesc committed:
Remove unnecessary replace

vanruesc committed:
Update pnpm-lock.yaml

vanruesc committed:
Version 1.3.2

Zosoled committed:
Use JS loader and export contents.
Line breaks should be retained when either (a) not minifying, or (b) preserving legal comments. This switches the esbuild loader to `"js"` and creates the returned contents a template literal instead of string. It also updates tests to conform to the template literal implementation.

Zosoled commented:
I hope you don't mind all the PRs; these are just things I have found useful to me as I continue my own work. :)

Zosoled committed:
Defer to esbuild initial options.
Users can already specifiy `minify` and/or `legalComments` in their esbuild config. To reduce the amount of setup they have to do, this changes the plugin options to defer to esbuild options, if specified, for minification and legal comment preservation.

vanruesc commented:
> I hope you don't mind all the PRs

Not at all, I appreciate the contributions :+1:

vanruesc commented:
> This switches the esbuild loader to "js" and creates the returned contents as a template literal instead of string.

I'm ok with this change, but I wonder if it has any impact on performance for projects where a lot of shader snippets are being imported. I'll do a quick check..

Zosoled committed:
Apply suggestions from code review

Use plain text instead of @defaultValue attributes to describe default behavior of parameters.

Co-authored-by: Raoul v. R. <vanruesc@users.noreply.github.com>

vanruesc commented:
> I'll do a quick check..

There's no difference in performance as far as I can tell, but the project I tested with "only" has 97 shader imports. It's not a good indicator, but it's still good to know that it doesn't slow down at all.

Zosoled commented:
Merged from main to resolve conflicts.
